src/lidar/read_hokuyo_lidar.cpp

```diff
@@ -0,0 +1,38 @@
+
+#include "PointCloudProcessing.h"
+#include "PointGenerator.h"
+#include "URGLidar.h"
+#include "../simulator/world_interface.h"
+#include "../simulator/utils.h"
+
+bool lidar_initialized = false;
+URGLidar urg_lidar;
+
+points_t readLidarScan() {
```

Comment:
I suppose this goes for all the sensors, but could we include an out parameter, like a pointer or reference, that indicates if the returned data is fresh? Otherwise we could read the same data in consecutive loops and mistake it to be different samples.

If we want to keep the method signature the same for this method we could make the parameter have a default value like `bool *fresh=nullptr`.

Reply:
That's an interesting point. I don't actually know yet how to check whether the URG sensor is returning fresh data or not. I'll add this as a Clickup task so we remember to look into it after virtual URC: https://app.clickup.com/t/vn739w